capstone/src/main/java/com/google/sps/search/Trie.java

```diff
@@ -93,6 +96,140 @@ public void insert(String name, String fullName) {
+   * the trie, then the name will be added to the set. Boolean reversed indicates whether to search
+   * for the name but first and last name are reversed. For example, "sweetandrew" is reversed and
+   * it can be split but into "Sweet Andrew," for which we can check that the reverse ("Andrew
+   * Sweet") is in the trie.
+   *
+   * @param input user input
+   * @param reversed boolean representing whether name is reversed
+   * @return Set of name suggestions
+   */
+  public Set<String> whitespace(String input, boolean reversed) {
+    Set<String> fullNames = new TreeSet<String>();
+
+    for (int i = 0; i < input.length(); i++) {
+      String firstName = input.substring(0, i);
+      String lastName = input.substring(i);
+      String fullName = "";
+      if (reversed) {
+        fullName = lastName + " " + firstName;
+      } else {
+        fullName = firstName + " " + lastName;
+      }
+      if (this.hasFullName(firstName, fullName)) {
+        fullNames.add(fullName);
+      }
+    }
+    return fullNames;
+  }
+
+  /**
+   * Finds names that are within the maximum LED distance of input by calling a recursive helper
+   * method that uses dynamic programming to find the edit distance between words.
+   *
+   * @param input user input
+   * @return Map from possible names to its edit distance from input
+   */
+  public Map<String, Integer> searchLed(String input) {
+    input = input.toUpperCase();
+    Map<String, Integer> suggestionsMap = new HashMap<String, Integer>();
+    int inputLength = input.length();
+    List<Integer> currRow = new ArrayList<Integer>();
+
+    for (int i = 0; i <= inputLength; i++) {
+      currRow.add(i);
+    }
+
+    Map<String, Integer> suggestedNames = new HashMap<String, Integer>();
+    for (Character c : this.children.keySet()) {
+      this.searchLedRecursive(this.children.get(c), c, input, currRow, suggestionsMap);
+      suggestedNames.putAll(suggestionsMap);
+    }
+
+    return suggestedNames;
+  }
+
+  /**
+   * Recursive helper method that populates the new row with values indicating edit distance with
+   * input name. These values are based off of the subproblem (the review row that is passed in).
+   *
+   * @param currNode current trie
+   * @param currChar current character
+   * @param input user input
+   * @param prevRow previous row in matrix
+   * @param suggestions map of suggested names
+   */
+  private void searchLedRecursive(
+      Trie currNode,
+      Character currChar,
+      String input,
+      List<Integer> prevRow,
+      Map<String, Integer> suggestions) {
+    int numColumns = input.length() + 1;
+    List<Integer> currRow = new ArrayList<Integer>();
+    currRow.add(prevRow.get(0) + 1);
+
+    for (int j = 1; j < numColumns; j++) {
+      if (input.charAt(j - 1) == currChar) {
+        currRow.add(prevRow.get(j - 1));
+      } else {
+        currRow.add(1 + Math.min(Math.min(prevRow.get(j), currRow.get(j - 1)), prevRow.get(j - 1)));
```

Comment:
`Math.min(Math.min(prevRow.get(j), currRow.get(j - 1)` can you do this one line above, and give it a good variable name? Hard for me to follow as is.